experimental/GroebnerWalk/src/markedGB.jl

```diff
@@ -0,0 +1,121 @@
+using Oscar 
+import Oscar: gens, base_ring
+import Base: length, getindex, setindex!
+
+#constructs a "markedGB" consisting of two lists 
+#gens, a vector of generators of a Gröbner basis
+#markings, a vector "Markings" of G, corresponding to leading terms w.r.t some monomial ordering
+#TODO: maybe initialize s.t. G is reduced/monic? 
+
+struct MarkedGroebnerBasis
+    gens::Vector{<:MPolyRingElem}
+    markings::Vector{<:MPolyRingElem}
```

Comment:
do both of these vectors contain elements over the same mpoly ring? If yes, it would be very beneficial to add a type parameter to `MarkedGroebnerBasis` and then use this to make these fields of type `Vector{T}`. This helps julia very much to infer types and make the resulting code more efficient